cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,84 @@
 /// <reference types='cypress' />
 
 describe('Student Registration page', () => {
-  before(() => {
+  let user;
 
+  before(() => {
+    cy.visit('/');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
   });
 
-  it('', () => {
+  it('should register a new student', () => {
+    cy.findByPlaceholder('First Name')
+      .type(user.firstName);
+
+    cy.findByPlaceholder('Last Name')
+      .type(user.lastName);
+
+    cy.get('#userEmail').type(user.email);
+
+    cy.contains('.custom-control-label', user.gender)
+      .click();
+
+    cy.findByPlaceholder('Mobile Number')
+      .type(user.mobileNumber);
+
+    cy.get('#dateOfBirthInput')
+      .type('{selectAll}15 Sep 2005{enter}');
```

Comment:
The date of birth is hard-coded in the test which is not a best practice. It would be better if the date of birth was a property on the user object so it can be changed easily and dynamically.